### ISSUE TITLE
Facility Locator: Error state on input fields should persist regardless of focus

### ISSUE THREAD

**Issue by randimays** [department-of-veterans-affairs/va.gov-cms#20271]
## Description

On the Facility Locator, when there is an error state on a field because the value is missing or invalid, the error message should persist regardless of what the user is doing with the other fields. In some scenarios, the error bounces back and forth between two fields.

https://github.com/user-attachments/assets/f8edb388-e9f1-45d3-bdb4-4b2cdc75df92

**This ticket needs Product, Design and Accessibility confirmation that:** if a field has been interacted with and does not have a valid value, the error state should be visible and continue until the user fixes the error.

### User story
As a Veteran using facility locator to find a VA facility for a specific service, in order to not get confused with the search form interaction and get accurate search results, I would like to know which search form fields contain errors until I edit the fields and resubmit the search.

## Acceptance criteria

- [ ] Error state persists on all search form fields that contain errors until the field input is revised and resubmitted
- [ ] For fields with error status, set focus on input (not select wrapper), see [comment in 22523](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/22523#issuecomment-3604118860) 
- [ ] For testing purposes, focus should move to the first field containing error upon submit (related ticket: #22523)

## Definition of done
- [x] Code completed; PR created
- [x] Review instance created/available; internal team notified to start reviews
- [ ] Automated tests passed 
- [ ] Code review (meets established quality and security standards)
- [ ] PR approved
- [ ] Design review (meets design & UX requirements)
- [ ] A11y review (meets accessibility standards)
- [x] Functional/feature review (meets acceptance criteria)
- [ ] Testing completed; dev team notified to deploy
- [ ] PO/PM/external verify in prod
- [ ] Documentation updated

**Comment by ACParker89** [comment]
Blocked due to #20370

**Comment by qjac** [comment]
@bryan-thompsoncodes and @r-bartlett-gsa, I just tested the form in Edge through CAG. [This is the review instance used.](http://d214804d8ebfdfb60041c5cd66e0fdf1.vfs.va.gov/find-locations)

And I'm seeing errors not triggering correctly. [See video](https://github.com/user-attachments/assets/a9155687-d9bd-4009-b45c-92f8182b4c76).
- Navigate to facility locator and hit search without interacting with the form.
- Note that only the location field errors, even though both location and facility locator are required.
- Then tab to next field (without entering a location).
- Note that the location field loses it's error message even though it remains empty.
**Expected behavior:** Any fields with an error, error on submit. And once those errors are shown, they persist until corrected.

I didn't check all of the form field combos, but I am also seeing the Community provider Service type field not erroring when it's empty and required. Screenshot is after clicking search with only the facility type field filled.
<img width="348" height="566" alt="Image" src="https://github.com/user-attachments/assets/84a811e3-5edf-46d3-88fa-8d40ec450348" />

**Comment by r-bartlett-gsa** [comment]
@bryan-thompsoncodes Following up on this. I will assign this ticket to you, given that you already worked on it (https://github.com/department-of-veterans-affairs/vets-website/pull/40384) and Jacqueline's comment above (https://github.com/department-of-veterans-affairs/va.gov-cms/issues/20271#issuecomment-3721144054)
